File: app/src/main/java/com/students/studensattendance/RetrofitRequest.java
package com.students.studensattendance;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitRequest {
    public static String Base_url="https://demo.todesign.in/admin/api/";
   public static Retrofit retrofit;
    public static Retrofit getInstance(){
        if(retrofit==null){
            retrofit=new Retrofit.Builder()
                    .baseUrl(Base_url)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }
}

File: app/src/main/java/com/students/studensattendance/apirequest/RetrofitRequest.kt
package com.students.studensattendance.apirequest

import com.students.studensattendance.model.Student
import com.students.studensattendance.model.StudentDetail
import retrofit2.Call
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.POST

interface RetrofitRequest {
    @POST("students")
    fun getStudents(): Call<Student>

    companion object{
        val base_url: String = "https://demo.todesign.in/admin/api/"
        var retrofitRequest: RetrofitRequest?=null
        fun getInstance() : RetrofitRequest{
           if (retrofitRequest==null){
               val retrofit = Retrofit.Builder()
                   .baseUrl(base_url)
                   .addConverterFactory(GsonConverterFactory.create())
                   .build()
               retrofitRequest = retrofit.create(RetrofitRequest::class.java)
           }
            return retrofitRequest!!
        }

    }
}